src/cljs/org/broadinstitute/firecloud_ui/main.cljs

```diff
@@ -126,24 +135,24 @@
              [:div {}
               [TopNavBar {:selected-item page
                           :on-nav (fn [item] (nav/navigate (:nav-context props) (name item)))}]
-              (let [item (first (filter #(= (% :key) page) top-nav-bar-items))]
+              (let [item (first (filter #(= (% :key) page) routes))]
                 (if item
                   ((item :render) {:nav-context nav-context})
                   [:div {} "Page not found."]))]))]]))
    :component-did-mount
    (fn [{:keys [this state]}]
      (endpoints/profile-get
-      :isRegistrationComplete
-      (fn [{:keys [success? status-text get-parsed-response]}]
-        (cond
-          (and success? (= (get-in (get-parsed-response) ["keyValuePair" "value"]) "true"))
-          (swap! state assoc :registration-status :registered)
-          (not success?)
-          (swap! state assoc :registration-status :not-registered)
-          :else
-          (do
-            (set! (.-errorMessage this) status-text)
-            (swap! state assoc :registration-status :error))))))})
+       (fn [{:keys [success? status-text get-parsed-response]}]
+         (let [parsed-values (common/parse-profile (get-parsed-response))]
+           (cond
+             (and success? (= (:isRegistrationComplete parsed-values) "true"))
+             (swap! state assoc :registration-status :registered :name (:name parsed-values))
+             (not success?)
+             (swap! state assoc :registration-status :not-registered)
```

Comment:
This would be good time to fix another bug: 404 means the key isn't set, but anything else is a server error. The `:else` condition here is never actually hit, so a server error sends the user to the registration page (which isn't right).